cypress/e2e/ui/Settings/Application-Settings/zone.cy.js

```diff
@@ -0,0 +1,320 @@
+    }
+    return null; // has no impact - just to get rid of eslint warning
+  });
+}
+
+describe('Automate Schedule form operations: Settings > Application Settings > Settings > Zones > Configuration > Add a new Zone', () => {
+  beforeEach(() => {
+    cy.login();
+    // Navigate to Application-Settings
+    cy.menu(settingsOption, appSettingsMenuOption);
+    // Expand Settings accordion panel if not already expanded
+    cy.accordion(settingsOption);
+    // Select "Zones" accordion item
+    cy.intercept('POST', /\/ops\/tree_select\?id=.*&text=.*/).as(
+      'treeSelectApi'
+    );
+    cy.selectAccordionItem([manageIQRegionAccordItem, zonesAccordItem]);
+    cy.wait('@treeSelectApi');
+  });
+
+  it('Validate the visibility and state of add form elements', () => {
+    // Open add form
+    cy.toolbar(configToolbarButton, addZoneConfigOption);
+    // Validate fields
+    validateFormElements();
+  });
+
+  it('Checking whether cancel button works on the add form', () => {
+    // Open add form
+    cy.toolbar(configToolbarButton, addZoneConfigOption);
+    // Cancelling the form
+    cy.getFormFooterButtonByType(cancelButton).click();
+    cy.expect_flash(flashTypeWarning, flashMessageOperationCanceled);
+  });
+
+  it('Checking whether add, edit & delete zone works', () => {
+    /* ===== Add ===== */
+    // Adding zone
+    addZone();
+    cy.expect_flash(flashTypeSuccess, flashMessageZoneUpdated);
+    // Expand Settings accordion panel if not expanded
+    cy.accordion(settingsOption);
+    /* ===== Edit ===== */
+    // Select the created zone
+    cy.selectAccordionItem([
+      manageIQRegionAccordItem,
+      zonesAccordItem,
+      `Zone: ${initialZoneDescription}`,
+    ]);
+    cy.wait('@treeSelectApi');
+    // Open edit form
+    cy.toolbar(configToolbarButton, editZoneConfigOption);
+    // Update IP & scan limit
+    cy.getFormInputFieldById(descriptionInputFieldId)
+      .clear()
+      .type(updatedServerIp);
+    cy.getFormSelectFieldById(maxScanSelectFieldId).select(updatedMaxScanLimit);
+    // Save the form
+    cy.getFormFooterButtonByType(saveButton, 'submit')
+      .should('be.enabled')
+      .click();
+    cy.expect_flash(flashTypeSuccess, flashMessageZoneUpdated);
+    /* ===== Delete ===== */
+    // Deleting the zone
+    cy.expect_browser_confirm_with_text({
+      confirmTriggerFn: () =>
+        cy.toolbar(configToolbarButton, deleteZoneConfigOption),
+      containsText: deleteConfirmText,
+    });
+    cy.expect_flash(flashTypeSuccess, deleteConfirmText);
+  });
+
+  it('Validate the visibility and state of edit form elements', () => {
+    // Adding zone
+    addZone();
+    // Expand Settings accordion panel if not expanded
+    cy.accordion(settingsOption);
+    // Select the created zone
+    cy.selectAccordionItem([
+      manageIQRegionAccordItem,
+      zonesAccordItem,
+      `Zone: ${initialZoneDescription}`,
+    ]);
+    cy.wait('@treeSelectApi');
```

Comment:
Is this the same treeSelectApi from the beforeEach?  I wonder if this can/should be moved into the selectAccordionItem in the future.  Our other functions seem to handle waits for us but this one doesn't.  It's stands out to me as odd.